### ISSUE TITLE
pygame camera not closed when frozen to exe using cx_freeze.

### ISSUE THREAD

**Issue by fusai19** [pygame/pygame#2878]
Below is simple code(camera.py), opening and closing camera.
```
import pygame
import pygame.camera
from time import sleep


pygame.camera.init()
pygame.camera.list_cameras() #Camera detected or not
cam = pygame.camera.Camera(0,(640,480))
cam.start()
img = cam.get_image()
pygame.image.save(img,r"sd.jpg")
cam.stop()

while 1:
    sleep(10)
```

when run using python camera.py, the camera is called soon after cam.stop() is executed.

However, when converted into exe file using cx_freeze,
the camera is still up.( The flash light near camera is up and is turned on, perhaps camera not closed) even after cam.stop() called.

**Comment by Starbuck5** [comment]
This depends a lot on your pygame version and OS, because that changes what camera backend it uses.

**Comment by fusai19** [comment]
When cx freeze is used, after the build, in /lib folder, it looks like VideoCapture is used, I believe it is the backend.

Nevertheless, when I forcefully set
```
cam = None
```
after cam.stop(), the camera is finally turned off. Not sure of why this weird behaviour, may be related to garbage collection.

**Comment by Starbuck5** [comment]
Tell me your pygame version and OS!

I recently added a get_backends() function, which will show you the backends it thinks it can use. So the default for your system would be the first element of that list.

**Comment by fusai19** [comment]
pygame version: 2.0.2
OS: Windows 10.
I guess I might be running older version as get_backends() returning attribute error on my machine.
As per documentation, pygame uses MSMF builtin windows camera, probably the issue could be with VideoCapture. I am sure on building with cx_freeze, pygame might have used VideoCapture as backend.
**I have seen no where any MSMF dll's that would other wise put by cx_freeze in /lib folder and the only option is VideoCapture as backend.(Just an assumption, I might be wrong)**

**Comment by Starbuck5** [comment]
What are you talking about with VideoCapture? The backend, or the opencv thing? Because I’m 99% sure it’s not running either. It’s just running my windows camera backend.

And there aren’t DLLs because the library is from the system, it’s built into the OS.

**Comment by fusai19** [comment]
Yes VideoCapture as backend. 
I have not used Opencv backend. (I have not installed it.)

Let me be clear, when cx_freeze is used, it will move dll's in /lib. For example, when pygame is build with cx_freeze dll's like
SDL2.dll, libpng, libjpeg, etc... dll's are moved into /pygame folder as shown in the image.

**Comment by Starbuck5** [comment]
As I said before; Microsoft Media Foundation (MSMF) is a system library, so an app using it doesn’t need to include the DLLs in its folder. They exist on the system.

That screenshot of vidcap only shows cpp files, and one of them is labeled python 2.5.

Update your pygame to latest and uninstall vidcap from your python installation.

**Comment by fusai19** [comment]
@Starbuck5 Thanks for your suggestions, Installting latest pygamr have fixed all issues.

To my great surprise, not just one issue, using version 2.0.2, lot of issues like camera freeze etc.. happening when build using cx_freeze and converted to executable.

Updating to 2.1.0 fixed all issues. Real culprit was VideoCapture.
Also as rightly said, it is not required to include dll's in the folder, it was my misconception.

Thanks.

**Comment by Starbuck5** [comment]
Nice!

Did you notice any differences between the new windows backend that you're using now and the old videocapture one?

**Comment by fusai19** [comment]
Honestly speaking a lot of difference.
With MSMF as backend now, the application became really smooth. 

Issues while using VideoCapture:

- When VideoCapture is used, and camera is already in use by another application, webcam.get_image() was blocking forever, without raising SystemError and the application got hanged.
- Camera resolution should be set explicitly like 640*480 etc... We need to do lot of trail errors to figure out exact resolution.
- Web camera is not closed even after calling webcam.stop()
- GUI interface with webcam frames becomes slightly hanged when using VideoCapture.


All of those are fixed by updating to 2.1.0.
I suspect lot more things are actually fixed in 2.1.0 which I haven't explored yet.

**Comment by Starbuck5** [comment]
I'm happy to hear it! Sounds like the MSMF backend is pulling its weight :smile:
